browser/ui/brave_browser_command_controller.cc

```diff
@@ -131,7 +132,9 @@ void BraveBrowserCommandController::InitBraveCommandState() {
   // to normal window in this case.
   const bool is_guest_session = browser_->profile()->IsGuestSession();
   if (!is_guest_session) {
-    UpdateCommandForBraveRewards();
+    if (!brave_rewards::IsDisabledByPolicy(browser_->profile()->GetPrefs())) {
```

Comment:
All of this logic including guest sessions is fragile and inconsistent with the checks in RewardsServiceFactory. We should just be checking to see if RewardsService is null or not here

Reply:
or alternatively expose a util method that is used by RewardsServiceFactory internally so we have one source of truth for when rewards is enabled or not

Reply:
Tried to streamline this in https://github.com/brave/brave-core/pull/14928/commits/0d8ec0ce92f67f3f6c437197efacfc5dd107d636, PTAL